From: Vicente Bergas <vicencb@gmail.com>
Date: Mon, 8 Aug 2016 13:31:43 +0200
Subject: [PATCH] fix ARMv8 interference with ARMv7

CONFIG_CPU_V8 is checked against 'y' and 'n', but the case when the
variable is unset is not considered.
This patch only checks the variable against a single value 'y' so the
logic is always coherent even when the variable is unset.

Signed-off-by: Vicente Bergas <vicencb@gmail.com>
Tested-by: Vicente Bergas <vicencb@gmail.com>
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 arch/arm/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 9fc3cd3debaa..1ab35b9667e9 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -21,7 +21,7 @@ endif
 # at least some of the code would be executed with MMU off, lets be
 # conservative and instruct the compiler not to generate any unaligned
 # accesses
-ifeq ($(CONFIG_CPU_V8),n)
+ifneq ($(CONFIG_CPU_V8),y)
 CFLAGS += -mno-unaligned-access
 endif
 
